homeassistant/components/eheimdigital/climate.py

```diff
@@ -0,0 +1,139 @@
+"""EHEIM Digital climate."""
+
+from typing import Any
+
+from eheimdigital.heater import EheimDigitalHeater
+from eheimdigital.types import EheimDigitalClientError, HeaterMode, HeaterUnit
+
+from homeassistant.components.climate import (
+    PRESET_NONE,
+    ClimateEntity,
+    ClimateEntityFeature,
+    HVACAction,
+    HVACMode,
+)
+from homeassistant.const import (
+    ATTR_TEMPERATURE,
+    PRECISION_HALVES,
+    PRECISION_TENTHS,
+    UnitOfTemperature,
+)
+from homeassistant.core import HomeAssistant
+from homeassistant.exceptions import HomeAssistantError
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+
+from . import EheimDigitalConfigEntry
+from .const import HEATER_BIO_MODE, HEATER_PRESET_TO_HEATER_MODE, HEATER_SMART_MODE
+from .coordinator import EheimDigitalUpdateCoordinator
+from .entity import EheimDigitalEntity
+
+# Coordinator is used to centralize the data updates
+PARALLEL_UPDATES = 0
+
+
+async def async_setup_entry(
+    hass: HomeAssistant,
+    entry: EheimDigitalConfigEntry,
+    async_add_entities: AddEntitiesCallback,
+) -> None:
+    """Set up the callbacks for the coordinator so climate entities can be added as devices are found."""
+    coordinator = entry.runtime_data
+
+    async def async_setup_device_entities(device_address: str) -> None:
+        """Set up the light entities for a device."""
+        device = coordinator.hub.devices[device_address]
+
+        if isinstance(device, EheimDigitalHeater):
+            async_add_entities([EheimDigitalHeaterClimate(coordinator, device)])
+
+    coordinator.add_platform_callback(async_setup_device_entities)
+
+    for device_address in entry.runtime_data.hub.devices:
+        await async_setup_device_entities(device_address)
```

Comment:
One thing for the future, we should avoid calling `async_add_entities` a lot, so rather for first setup we batch them

Reply:
Okay, then I have to think about how to return the entities for the first setup and calling `async_add_entities` directly when founding devices on runtime.

Reply:
I can add a boolean parameter to the function, so that it either returns the entities (for the first setup) or calls `async_add_entities` directly (when a device is detected dynamically)

Reply:
I will do this after this PR is merged.